django/db/models/query.py

```diff
@@ -338,7 +338,22 @@ def __or__(self, other):
     ####################################
 
     def _iterator(self, use_chunked_fetch, chunk_size):
-        yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
+        clone = self._chain()
+
+        iterator = iter(clone._iterable_class(clone, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size))
+        while True:
+            clone._result_cache = list(islice(iterator, chunk_size))
+            clone._prefetch_done = False
+
+            if len(clone._result_cache) == 0:
+                clone._result_cache = None
+                return
+
+            if clone._prefetch_related_lookups:
```

Comment:
This conditional clause can be dropped once we perform an initial `not self._prefetch_related_lookups` as described above.